As upstream notes:

# Note: This test fails on Linux with musl libc versions that don't support
# the BIG5 encoding in 'iconv'.

diff --git a/gettext-tools/tests/msgconv-2 b/gettext-tools/tests/msgconv-2
index d286cda..d96c487 100755
--- a/gettext-tools/tests/msgconv-2
+++ b/gettext-tools/tests/msgconv-2
@@ -7,6 +7,8 @@
 # Note: This test fails on Linux with musl libc versions that don't support
 # the BIG5 encoding in 'iconv'.
 
+Exit 77
+
 cat <<\EOF > mco-test2.po
 # Chinese translation for GNU gettext messages.
 #
diff --git a/gettext-tools/tests/msgmerge-compendium-6 b/gettext-tools/tests/msgmerge-compendium-6
index 0afbe1e..1b2fe6b 100755
--- a/gettext-tools/tests/msgmerge-compendium-6
+++ b/gettext-tools/tests/msgmerge-compendium-6
@@ -10,6 +10,8 @@
 # Note: This test fails on Linux with musl libc versions that don't support
 # the EUC-KR encoding in 'iconv'.
 
+Exit 77
+
 : ${MSGCONV=msgconv}
 ${MSGCONV} --to-code=UTF-8 -o mm-ko.utf-8.pot "$wabs_srcdir"/mm-ko.ascii.pot
 
diff --git a/gettext-tools/tests/xgettext-python-3 b/gettext-tools/tests/xgettext-python-3
index ca0926d..2c49fd7 100755
--- a/gettext-tools/tests/xgettext-python-3
+++ b/gettext-tools/tests/xgettext-python-3
@@ -6,6 +6,8 @@
 # Note: This test fails on Linux with musl libc versions that don't support
 # the EUC-JP encoding in 'iconv'.
 
+Exit 77
+
 cat <<\EOF > xg-py-3a.py
 #!/usr/bin/env python
 # TRANSLATORS: Franè´Æois Pinard is a hero.
